pkg/kubelet/lifecycle/predicate.go

```diff
@@ -172,3 +175,50 @@ func removeMissingExtendedResources(pod *v1.Pod, nodeInfo *schedulernodeinfo.Nod
 	}
 	return podCopy
 }
+
+// setNodeAllocatableScalarResources scans pod requested scalar resources, and if necessary,
+// updates allocatable Resource in nodeInfo to at least equal to
+// the requested capacity. This allows pods that have already been scheduled on
+// the node to pass GeneralPredicates admission checking.
+func setNodeAllocatableScalarResources(nodeInfo *schedulercache.NodeInfo, admitPod *v1.Pod) {
+	var newAllocatableResource *schedulercache.Resource
+	var needUpdate bool = false
+	allocatableResource := nodeInfo.AllocatableResource()
+	if allocatableResource.ScalarResources == nil {
+		allocatableResource.ScalarResources = make(map[v1.ResourceName]int64)
+	}
+
+	ScalarResources := predicates.GetResourceRequest(admitPod).ScalarResources
```

Comment:
Nit: Can you share more about why we deviated from the `camelCase` naming convention here?

Reply:
so sorry, it's my fault, it should be "scalarResources", i will fix it immediately